internal/storage/storage.go

```diff
@@ -315,3 +320,29 @@ func (db *Database) GetAllURLForUser(user string) ([]middleware.JSONStructForAut
 	fmt.Sprintln(JSONStructList)
 	return JSONStructList, returnErr
 }
+
+func (db *Database) SearchID(url string) (int, error) {
+	var id int
+	query := fmt.Sprintf("select id from %s.%s where full_url = '%s'", schema, table, url)
```

Comment:
Вот так делать нельзя. Можно пробрасывать в query название таблицы и схемы, но вот запрос от пользователя необходимо экранировать. XSS-атаки - одни из самых разрушительных.
```suggestion
	query := fmt.Sprintf("select id from %s.%s where full_url = $1", schema, table)
```